build.gradle

```diff
@@ -99,25 +99,30 @@ dependencies {
 
 bootRun {
     sourceResources sourceSets.main
-}
 
-task runLocal {
-    dependsOn 'withH2'
-    dependsOn 'bootRun'
-    group 'application'
-    description 'Runs this project as a Spring Boot application using in-memory (H2) database'
     gradle.taskGraph.whenReady { graph ->
-        if (graph.hasTask(runLocal)) {
+        boolean dbDeclared = false;
+
+        graph.getAllTasks().forEach { task->
+            if(task.name.contains('with')){
+                dbDeclared=true;
+            }
+        }
+
+        if(!dbDeclared){
             logger.lifecycle('>>> Setting spring.profiles.active == "local" <<<')
             bootRun {
+                dependencies {
+                    implementation 'com.h2database:h2:1.4.200'
+                }
                 args = ['--spring.profiles.active=local']
```

Comment:
This will override any other profiles the user may have passed in. (for example `SPRING_PROFILES_ACTIVE=local,unsecure ./gradlew bootRun`) Can you append the local profile instead of overriding?
We don't have additional profiles currently but overriding the profiles could make debugging difficult in the future.

Reply:
Good catch, I have added a check to make sure that no spring profile is set as well